### ISSUE TITLE
SAFE Browser appears not to support Promise.finally()

### ISSUE THREAD

**Issue by theWebalyst** [maidsafe/safe_browser#807]
## Prerequisites

- [ ] Using yarn
- [x] Using an up-to-date master branch
- [ ] Link to stacktrace in a Gist (for bugs)

## Expected Behavior

Major browsers support `Promise.finally()` so SAFE Browser should also, as in:

```
safeApp.auth.loginFromUri(authUri).then(() => {
      console.log('loginFromUri')
  }).catch((e) => {
      console.log('catch: ', e)
  }).finally(() => {
      console.log('finally!!!')
  })
```

## Current Behavior

Using `Promise.finally()` such as in the above example results in an error:

```
Uncaught (in promise) TypeError: safeApp.auth.loginFromUri(...).then(...).catch(...).finally is not a function
```

**UPDATE:** as a test I have tried this with the completed SAFE Web App Tutorial example (in Vue), again chaging the `safeApp.auth.loginFromUri(authUri)` line to the above code fragment, and I get the same error.

So the issue is not restricted to my version of the tutorial implemented using svelte, and suggests it is a problem with the Promise implementation exposed to web apps by SAFE Browser.
## Possible Solution

This might be because the browser JavaScript engine doesn't support `Promise.finally()` or another possible cause is if a polyfill (perhaps from a dependency) is overwriting the browser implementation.

## Your Environment

- Node version : 10
- Version or Branch used : 0.13 dev build
- Operating System and version : Ubuntu 18.10
- Link to your project : https://github.com/theWebalyst/safetutorial-svelte
I've modified the file `safe-api.js` to use the code fragment above instead of the following line:

https://github.com/theWebalyst/safetutorial-svelte/blob/master/src/safe-api.js#L15

**Comment by theWebalyst** [comment]
UPDATE: as a test I have tried this with the completed SAFE Web App Tutorial example (in Vue), again chaging the `safeApp.auth.loginFromUri(authUri)` line to the above code fragment, and I get the same error.

So the issue is not restricted to my version of the tutorial implemented using svelte, and suggests it is a problem with the Promise implementation exposed to web apps by SAFE Browser.

**Comment by theWebalyst** [comment]
UPDATE: I'm a bit puzzled by this issue because `Promise.finally()` has been supported since at least Chrome v63, and if I understand correctly, v4.x of Electron which SAFE Browser v0.14 uses, has more recent Chromium than v63.

**Comment by theWebalyst** [comment]
Closing because I believe this is fixed in SAFE Browser 0.15.1 (presumably by the update of electron).

**Comment by joshuef** [comment]
Thanks @theWebalyst !
